[Profiler] Enable SOFT_ASSERT to log Invariant Violation to Kineto

aaronenyeshi opened the pull request:
Summary: Record the Soft assert to Kineto.

Test Plan: Internal CI Tests.

Differential Revision: D42219145

Pulled By: aaronenyeshi

robieta commented on torch/csrc/profiler/util.cpp:
```diff
@@ -97,6 +97,36 @@ bool softAssertRaises() {
+    const char* args) {
+#ifdef USE_KINETO
+  std::stringstream error;
+  error << cond << " SOFT ASSERT FAILED at " << file
```
nit: for future reference this is a good place to use `fmt`. (But not worth changing since you have CI mostly done.)

aaronenyeshi replied:
Great point! changing it to use fmt!

robieta reviewed (approved):
Awesome!

aaronenyeshi committed:
[Profiler] Enable SOFT_ASSERT to log Invariant Violation to Kineto (#92872)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/92872

Record the Soft assert to Kineto.

Test Plan: Internal CI Tests.

Reviewed By: fengxizhou, robieta

Differential Revision: D42219145

Pulled By: aaronenyeshi

fbshipit-source-id: 2dbd2c79033587fb70d3901667387e00746e8d8d
